packages/api/src/controllers/helpers.ts

```diff
@@ -771,45 +771,49 @@ export async function addDefaultProjectId(
     }
   };
 
-  const enrichResponseWithUserProjectId = (document) => {
+  const enrichResponseWithUserProjectId = async (document) => {
     if ("id" in document && "userId" in document && "user" in document) {
-      if (
-        (!document.projectId || document.projectId === "") &&
-        document.user?.defaultProjectId
-      ) {
-        document.projectId = document.user.defaultProjectId;
+      if (!document.projectId || document.projectId === "") {
+        if (document.user?.defaultProjectId) {
+          document.projectId = document.user.defaultProjectId;
+        } else {
+          let user = await db.user.get(document.userId, { useCache: true });
+          if (user.defaultProjectId) {
+            document.projectId = user.defaultProjectId;
+          }
```

Comment:
It's not really a problem if `defaultProjectId` is not set, right? We'd just be setting the `projectId` to empty or undefined, which it actually is anyway (if it has anything else it's a mistake!)

Not a problem checking it either, just feels like unnecessary branches to be reasoning about in the code, apart from the required ones.

```suggestion
          document.projectId = user.defaultProjectId;
```